File: CS/UserViewVariants.Web/WebUserViewVariantsController.cs
using UserViewVariants;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Web;

namespace UserViewVariants.Web {
    public class WebUserViewVariantsController : UserViewVariantsController {
        protected override void UpdateCurrentViewVariant(string selectedViewId) {
            base.UpdateCurrentViewVariant(selectedViewId);
            ((WebWindow)Application.MainWindow).RegisterStartupScript(Name, "window.location.reload();");
        }
        protected override void OnActivated() {
            base.OnActivated();
            UserViewVarintsAction.Active["ActiveInListViewOnly"] = View is ListView;
        }
    }
}

File: VB/UserViewVariants.Web/WebUserViewVariantsController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Web

Namespace UserViewVariants.Web

    Public Class WebUserViewVariantsController
        Inherits UserViewVariantsController

        Protected Overrides Sub UpdateCurrentViewVariant(ByVal selectedViewId As String)
            MyBase.UpdateCurrentViewVariant(selectedViewId)
            CType(Application.MainWindow, WebWindow).RegisterStartupScript(Name, "window.location.reload();")
        End Sub

        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            UserViewVarintsAction.Active("ActiveInListViewOnly") = TypeOf View Is ListView
        End Sub
    End Class
End Namespace
